From a4985a8eeca0fb69f55e3309fd084c0fee228b4b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Pavol=20=C5=BD=C3=A1=C4=8Dik?= <pzacik@redhat.com>
Date: Mon, 11 May 2026 07:17:34 +0200
Subject: [PATCH] Fix OpenSSL 4.0 compatibility

Replace direct ASN1 structure field access with ASN1_STRING_get0_data() and ASN1_STRING_length() accessor functions.
---
 xio-openssl.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/xio-openssl.c b/xio-openssl.c
index 344709b..e328ee2 100644
--- a/xio-openssl.c
+++ b/xio-openssl.c
@@ -1898,8 +1898,8 @@ static int openssl_handle_peer_certificate(struct single *sfd,
 		  case GEN_IPADD:
 		     {
 			/* binary address format */
-			const unsigned char *data = pName->d.iPAddress->data;
-			size_t len = pName->d.iPAddress->length;
+			const unsigned char *data = ASN1_STRING_get0_data(pName->d.iPAddress);
+			size_t len = ASN1_STRING_length(pName->d.iPAddress);
 			char aBuffer[INET6_ADDRSTRLEN]; 	/* canonical peername */
 			struct in6_addr ip6bin;
 
-- 
2.53.0

